### ISSUE TITLE
Nextcloud 11 overview

### ISSUE THREAD

**Issue by MorrisJobke** [nextcloud/server#745]
## Merged
- [x] Follow on twitter (@karlitschek @jospoortvliet #816)
- [x] User mentions in comments #753 (@blizzz) 
- [x] 4 byte support - upgrade path!!! (@nickvergessen) (https://github.com/nextcloud/server/issues/1560)
- [x] Appstore (@bernhardposselt)
  - [x] Nextcloud integration (@lukasreschke)
- [x] Nextcloud organization announcement (@nickvergessen) https://github.com/nextcloud/nextcloud_announcements
- [x] Share by email (@schiessle) **PR:** #657
- [x] Multi bucket (@icewind1991 / @rullzer) #1084
- [x] More activities (@nickvergessen)
  - [x] Move and rename of files #1418 + nextcloud/activity#43
  - [x] Calendars **PR:** #1733
  - [x] Events **PR:** #1733
  - [x] Todos **PR:** #1733
- [x] Move file/folder (@icewind1991) **PR:** #1754
- [x] Share current folder (@christophwurst) **PR:** #1618
- [x] Shared previews (@rullzer) **PR:** #1741
- [x] Kerberos & ADFS support (@LukasReschke)
- [x] App password permission control (@icewind1991) **PR:** #719
- [x] S3 (@icewind1991) **PR:** #768 
- [x] Password confirmation for several admin/personal settings (@nickvergessen) **PR:** #1447
- [x] CLI updater (@morrisjobke) **PR:** nextcloud/updater#45
- [x] Opt-In LDAP change password ability (@GitHubUser4234) **PR:** #1715
- [x] nextcloud_announcements needs URL adjustment for the final route (@nickvergessen) https://github.com/nextcloud/nextcloud_announcements/pull/11
- [x] Lookup server (@schiessle) (client part: #1946)
- [x] spreed.me (@icewind1991 @leonklingele @ivansss) #828 & https://github.com/strukturag/nextcloud-spreedme/issues/36
- [x] sharing
  - [x] Recognize remote users for activities (@nickvergessen) [nextcloud/activity#25](nextcloud/activity#25)
- [x] documentation:
   - [x] mysql 4 byte support & info in admin panel (@nickvergessen)
   - [x] ADFS & Kerberos (@lukasreschke)
- [x] enable nextcloud_announcements by default (@nickvergessen)
- [x] password confirmation layout (@jancborchardt @christophwurst)
- [x] sign the content of the nextcloud_announcements response & only show trusted ones (@lukasreschke)
- [x] update URL of nextcloud_announcements (@nickvergessen) 
- [x] cleanup old previews (@rullzer) #2152

## ToDo after review on 2016-11-15
- [ ] documentation:
   - [ ] CLI updater & general updater overhaul (@morrisjobke)
- [ ] app store - ratings (official, supported, ...) how to differentiate (@lukasreschke @jospoortvliet)
- [ ] Release polishing (overview: https://github.com/nextcloud/server/issues/2134)

## Not finished - delayed to 12
- [ ] Undo delete (@christophwurst) **PR:** #1610
- [ ] Opt-In LDAP password expiry handling (@GitHubUser4234) **PR:** #1023 (blocked by https://github.com/nextcloud/server/issues/1303)
- [ ] Sharepoint (@blizzz)
- [ ] Introduce timeline (@christophwurst)
- [ ] Skip version upgrades (@lukasreschke @morrisjobke)
- [ ] Signature panel ("I have seen it") (@blizzz) #1213
- [ ] Next generation federeated sharing (@schiessle / @rullzer)
- [ ] @mentioning auto-complete #2443

This is a rough list of topics we want to work on for Nextcloud 11. It lists the topic and a person who want to work on it and manage the tasks that are to be done. This is up to change and the exact specs are not set in stone now. Some of these are tiny, others are huge - we'll create issues to explain and expand on them the coming week.

If anybody is interested to help out for a specific topic just contact the listed person 😃 And of course if you want to work on something different: feel free to do so and add yourself to the list 😄 

Pls be sure to add a issue number of PR link to what you do/did.

**Comment by BernhardPosselt** [comment]
Do we need to check off anything since you've added check boxes?

**Comment by blizzz** [comment]
@BernhardPosselt when it's done

**Comment by MorrisJobke** [comment]
> @BernhardPosselt when it's done

Or when it's transferred into an issue ;)

**Comment by bonanza123** [comment]
Is there an approx. ETA? 
As I couldn't find a milestone for NC 10, does it mean it will be skipped, i.e. directly from 9.x to 11?

**Comment by BernhardPosselt** [comment]
@bonanza123 10 is due in a few days to weeks

ETA is when it's done, but new releases usually take around 4-12 months

**Comment by jospoortvliet** [comment]
10 is planned for the week of 22-27 of August. 11 is December timeframe.

**Comment by Bugsbane** [comment]
I'd love to see more of these linked to issue numbers. I tried searching for a few to add myself, without much luck.

**Comment by jospoortvliet** [comment]
@Bugsbane that will be done this week - we largely have to create those issues still ;-)

**Comment by pierrejochem** [comment]
@blizzz sharepoint: will it be opensource?

**Comment by rullzer** [comment]
@pierrejochem yes

**Comment by pierrejochem** [comment]
👍

**Comment by afunix** [comment]
What about client side encryption?
Don't think it can be called secure, if a kernel/Apache vulnerability can reveal all clear data to an attacker.

**Comment by LukasReschke** [comment]
> What about client side encryption?
Don't think it can be called secure, if a kernel/Apache vulnerability can reveal all clear data to an attacker.

You're welcome to help contributing to the client side encryption. Please also note that this ticket is not a "I wish features" list. For that one please create a new issue.

**Comment by BernhardPosselt** [comment]
@afunix you can use client side encryption today! Just encrypt your file before uploading it. Or dont upload it at all if its that important ;)

Apart from that this is more of a sync client ticket and AFAIK someone began to work on it already however dont expect anything in the near future. If you want this feature then I'd suggest you contact the other guy and start programming ;)

**Comment by blizzz** [comment]
@pierrejochem that'll be awesome! Let's discuss once 10 is out.

**Comment by pierrejochem** [comment]
@blizzz Yes, that sounds like a good plan. Will it be implemented in the files_external app or a seperate one? Or open for discussion?

**Comment by blizzz** [comment]
It makes most sense to get this right into files_external, as all the components are already there and we just need to hook into it. Except there are strong arguments against it?

**Comment by pierrejochem** [comment]
@blizzz I'm with you so it had to be there in my opinion too

**Comment by Bugsbane** [comment]
I'd be interested in helping @nickvergessen with the announcement, if you're up for that, Nick. I have a background in copywriting.

**Comment by nickvergessen** [comment]
> I'd be interested in helping @nickvergessen with the release announcement, if you're up for that, Nick. I have a background in copywriting.

The task may be phrased a little strange, what we actually want to do is implement something like a "pull announcements from nextcloud.com/org to your instance". We will do this mostlikely on top of https://github.com/nextcloud/announcementcenter

**Comment by rullzer** [comment]
@Bugsbane but talk to @jospoortvliet :)

**Comment by Bugsbane** [comment]
> talk to @jospoortvliet :)

Will do. @jospoortvliet What's the best channel to reach you on?

**Comment by jospoortvliet** [comment]
email me on jos at the nextcloud.com server ;-)

Thanks for the offer, help would be awesome!

**Comment by jancborchardt** [comment]
Please only check off items when they are _done_, not when they just have their own issue. This is the overview issue.

**Comment by michi-roth** [comment]
ownCloud has some problems reliable detect changes in external storage.
https://github.com/owncloud/core/issues/11797

Are there any plans to address this in issue nextcloud?

**Comment by blizzz** [comment]
@icewind1991 ^

**Comment by blizzz** [comment]
> @blizzz sharepoint: will it be opensource? If yes, then I could support

> It makes most sense to get this right into files_external, as all the components are already there and we just need to hook into it. Except there are strong arguments against it?

@pierrejochem would you have some time next week to discuss this? Either on IRC or spreed.me. And, btw, are you coming to the conference?

**Comment by pierrejochem** [comment]
@blizzz yes, I have Time. Which Conference?

**Comment by mar1u5** [comment]
This one: https://conf.nextcloud.com @pierrejochem

**Comment by pierrejochem** [comment]
@MariusBluem I will ask my boss on monday

**Comment by pierrejochem** [comment]
@blizzz @MariusBluem I will come but only in week 38. 1-2 days only. I need to register right?

**Comment by pierrejochem** [comment]
@blizzz I'm available at IRC. My nickname is pierrejochem

**Comment by blizzz** [comment]
@pierrejochem Sorry, i managed to catch a flu (thought/hoped it would be a weekend thingy, but no) and need to postpone this unfortunately.  :mask: 

> I need to register right? Whats days fits best for you?

Yes, this is helpful. I be there at any day. Basically, the question what you also want to get out of it. [Friday](https://nextcloud.com/friday-focus-enterprise-use-of-nextcloud-at-the-conference/) is focused on enterprise use, Sat and Sun are the usual conference parts. These three days are mostly talks and talking (but also workhops), directly hands on hacking are Mon-Thu. For instance, you could take Sun+Mon to have a combination.

**Comment by pierrejochem** [comment]
@blizzz No problem, get well soon and I hope I can get the sunday. Have to ask my wife. Monday is no problem.

**Comment by BernhardPosselt** [comment]
App store is feature complete with regards to the 0.1.0 milestone and will enter the stabilization and translation period. The 0.1.0 deployment will kill the current database , so you will need to re-register. App devs can now also start to upload their own apps.

Will keep you updated once the release is deployed

**Comment by schiessle** [comment]
> The 0.1.0 deployment will kill the current database , so you will need to re-register. App devs can now also start to upload their own apps.
>
> Will keep you updated once the release is deployed

Isn't it better to wait until the release is deployed before creating accounts and uploading apps if the deployment will kill the current database? Or did I misunderstood something?

**Comment by LukasReschke** [comment]
Before wiping the DB we should also have https://github.com/nextcloud/appstore/issues/249. I'm looking into this…

**Comment by BernhardPosselt** [comment]
@schiessle like you said, app devs should be able to upload their apps **after** we kill the db :)

**Comment by BernhardPosselt** [comment]
Ok, certificates, signatures, documentation and validation is now live (including the db wipe). Just ready for the conf :sunglasses: 

I'd say we keep off publishing a release until 11 is out. We may still need to adjust the API depending on if people implementing the API in Nextcloud run into issues :)

**Comment by blizzz** [comment]
> @blizzz No problem, get well soon and I hope I can get the sunday. Have to ask my wife. Monday is no problem.

I think best is talking at the conf, after the week i am still busy with the conf and would not be so well prepared. Do you have fixed dates when you will be there?

**Comment by jancborchardt** [comment]
> Needs URL adjustment for the final route (@jospoortvliet)

Is this done @jospoortvliet @nickvergessen?

**Comment by jancborchardt** [comment]
> @mentioning auto-complete

@blizzz any chance we can get this? It’s pretty essential for the whole @mentioning and discoverability, especially as we mostly show real names instead of usernames :\

**Comment by nickvergessen** [comment]
> > Needs URL adjustment for the final route (@jospoortvliet)
>
> Is this done @jospoortvliet @nickvergessen?

It's done, once https://github.com/nextcloud/nextcloud_announcements/blob/master/lib/Cron/Crawler.php#L37 does **NOT** point to `https://nextcloud.com/blogfeed/` anymore.

**Comment by nickvergessen** [comment]
All the remaining stuff should go into 12, so moving this.

**Comment by nickvergessen** [comment]
Moved open tasks to https://github.com/nextcloud/server/issues/2672
